HomeWork/src/main/java/generics/Box.java

```diff
@@ -0,0 +1,101 @@
+package generics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+
+    private List<Fruit> fruitBox = new ArrayList<>();
+    private T fruit;
+
+    public Box(T fruit) {
+        this.fruit = fruit;
+    }
+
+
+    public T getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(T fruit) {
+        this.fruit = fruit;
+    }
+
+    public List<Fruit> getFruitBox() {
+        return fruitBox;
+    }
+
+    public void setFruitBox(List<Fruit> fruitBox) {
+        this.fruitBox = fruitBox;
+    }
+
+    public <O extends T> void add(O obj) {
+        Objects.requireNonNull(obj);
+        this.fruitBox.add(obj);
+    }
+
+    public boolean addAllFruit(List<? extends Fruit> fruitList) {
+        if (fruitList.isEmpty()) {
+            throw new NullPointerException("An empty list is passed to the method argument");
+        }
+        if (this.fruitBox.isEmpty()) {
+            if (containsListFruit(fruitList)) {
+                this.fruitBox.addAll(fruitList);
+                return true;
+            }
+        }
+        if (containsListFruit(fruitList) && fruit instanceof Orange) {
+            this.fruitBox.addAll(fruitList);
+            return true;
+        } else if (containsListFruit(fruitList) && fruit instanceof Apple) {
```

Comment:
Class `Box` should know nothing about the Orange or Apple class! You should check the type by comparing the class of the elements, like using `getClass`.